ext/enchant/enchant.c

```diff
@@ -462,44 +502,68 @@ PHP_FUNCTION(enchant_broker_request_dict)
 		RETURN_FALSE;
 	}
 
-	d = enchant_broker_request_dict(pbroker->pbroker, (const char *)tag);
-	if (d) {
-		pos = pbroker->dictcnt++;
-		if (pbroker->dictcnt) {
-			pbroker->dict = (enchant_dict **)erealloc(pbroker->dict, sizeof(enchant_dict *) * pbroker->dictcnt);
-		} else {
-			pbroker->dict = (enchant_dict **)emalloc(sizeof(enchant_dict *));
-			pos = 0;
-		}
-
-		dict = pbroker->dict[pos] = (enchant_dict *)emalloc(sizeof(enchant_dict));
-		dict->id = pos;
-		dict->pbroker = pbroker;
-		dict->pdict = d;
-		pbroker->dict[pos] = dict;
+	pdict = enchant_broker_request_dict(pbroker->pbroker, (const char *)tag);
+	if (pdict) {
+		pbroker->nb_dict++;
 
-		dict->rsrc = zend_register_resource(dict, le_enchant_dict);
-		GC_ADDREF(pbroker->rsrc);
-		RETURN_RES(dict->rsrc);
+		object_init_ex(return_value, enchant_dict_ce);
+		dict = Z_ENCHANT_DICT_P(return_value);
+		dict->pdict =pdict;
+		ZVAL_COPY(&dict->zbroker, broker);
 	} else {
 		RETURN_FALSE;
 	}
 }
 /* }}} */
 
+/* {{{ proto resource EnchantDict::__construct(resource broker, string tag)
+	create a new dictionary using tag, the non-empty language tag you wish to request
+	a dictionary for ("en_US", "de_DE", ...) */
+PHP_METHOD(EnchantDict, __construct)
+{
+	zval *broker;
+	enchant_broker *pbroker;
+	enchant_dict *dict;
+	EnchantDict *pdict;
+	char *tag;
+	size_t taglen;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "Os", &broker, enchant_broker_ce, &tag, &taglen) == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	PHP_ENCHANT_GET_BROKER;
+
+	if (taglen == 0) {
+		php_error_docref(NULL, E_WARNING, "Tag cannot be empty");
+		RETURN_FALSE;
```

Comment:
Cannot return false from constructor, this should be ValueError.

Reply:
fixed